File: Gombang/Weekly Contest 411/3258_Count_Substrings_That_Satisfy_K-Constraint_I.cs
public class Solution
{
	public int CountKConstraintSubstrings(string s, int k)
	{
		int result = 0;
		int leftIndex = 0;

		while (leftIndex < s.Length)
		{
			int zeroCnt = 0;
			int oneCnt = 0;

			for (int i = leftIndex; i < s.Length; i++)
			{
				if (s[i] == '0')
					zeroCnt++;
				else
					oneCnt++;

				if (zeroCnt <= k || oneCnt <= k)
					result++;
				else
					break;
			}

			leftIndex++;
		}

		return result;
	}
}

File: Lucy/Weekly Contest 411/3258_Count_Substrings_That_Satisfy_K-Constraint_I.js
/**
 * 3258. Count Substrings That Satisfy K-Constraint I
 * url: https://leetcode.com/problems/count-substrings-that-satisfy-k-constraint-i/
 *
 * topic: String, Sliding Window
 * difficulty: Easy
 * date: 2024.08.23(FRI)~
 */

/**
 * @param {string} s
 * @param {number} k
 * @return {number}
 */
var countKConstraintSubstrings = function (s, k) {
  const splited = s.split("");
  let cnt = 0;

  for (let window = 1; window <= splited.length; window++) {
    for (let i = 0; i < splited.length && i + window <= splited.length; i++) {
      const substring = splited.slice(i, i + window);

      const oneCnt = substring.filter((n) => n === "1").length;
      const zeroCnt = substring.filter((n) => n === "0").length;

      if (zeroCnt <= k || oneCnt <= k) cnt++;
    }
  }

  return cnt;
};

console.log(countKConstraintSubstrings("10101", 1));
console.log(countKConstraintSubstrings("1010101", 2));
console.log(countKConstraintSubstrings("11111", 1));
